src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -454,30 +474,46 @@ public Map<FingerprintIdDetails, Fingerprint> fingerprintVcf(final Path vcfFile)
     };
 
     /**
-     * Does same thing as {@link #fingerprintSamFile(Path, Function)} but in the old way that required that you pass in the loci of interest.
+     * Does same thing as {@link #fingerprintSamFile(Path, Path, boolean, Function)} but in the old way that required that you pass in the loci of interest.
      * Since the loci are always the same as in {@link #haplotypes} there's no need for this method signature anymore
      *
-     * @deprecated use {@link #fingerprintSamFile(Path, Function)} instead.
+     * @deprecated use {@link #fingerprintSamFile(Path, Path, boolean, Function)} instead.
      */
     @Deprecated
     public Map<FingerprintIdDetails, Fingerprint> fingerprintSamFile(final Path samFile, final IntervalList loci) {
-        return fingerprintSamFile(samFile, HaplotypeProbabilitiesFromSequence::new);
+        return fingerprintSamFile(samFile, null, false, HaplotypeProbabilitiesFromSequence::new);
+    }
+
+    private SamReader getSamReader(final Path samFile, final Path indexPath, final boolean forceIndex) {
+        final SamInputResource samResource = SamInputResource.of(samFile);
+        if (indexPath != null) {
+            samResource.index(indexPath);
```

Comment:
```suggestion
    samResource.index(indexPath, seekableChannelFunction);
```

Reply:
Thanks, just implemented this change and checked this should be equivalent to the old code. I left a comment pointing to the relevant issue in case someone tries to delete the `seekableChannelFunction` wrapper.